src/main/java/org/prebid/server/auction/InterstitialProcessor.java

```diff
@@ -0,0 +1,393 @@
+package org.prebid.server.auction;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.Banner;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Device;
+import com.iab.openrtb.request.Format;
+import com.iab.openrtb.request.Imp;
+import io.vertx.core.json.Json;
+import org.prebid.server.exception.InvalidRequestException;
+import org.prebid.server.proto.openrtb.ext.request.ExtDevice;
+import org.prebid.server.proto.openrtb.ext.request.ExtDeviceInt;
+import org.prebid.server.proto.openrtb.ext.request.ExtDevicePrebid;
+import org.springframework.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class InterstitialProcessor {
+
+    private static final int MAX_SIZES_COUNT = 10;
```

Comment:
As we don't have any dependencies outside can it be a part of `InterstitialSize` class?